### PR TITLE
john: apply suggestions

### PR DESCRIPTION
**Pull request by sbrl**
Ref #3964.

/cc @agnivade, @dhondta

### CODE REVIEW

File: pages/common/john.md [hunk 1]
```diff
@@ -25,12 +26,12 @@
 
 - Crack password hashes, using a specific hash format:
 
-`john --format={{md5crypt|...}} {{path/to/hashes.txt}}`
+`john --format={{hash_format}} {{path/to/hashes.txt}}`
```
Comment (zdroid) [inline pages/common/john.md:28]: What's the reason behind this change?

Comment (agnivade) [reply pages/common/john.md:28]: I actually just wanted to remove the `...` and not change the concrete example. My reasoning was that we use the `|` format to show alternative options, but only when they are complete. If all options cannot be specified, then it's simple to just use one option, as it's implicit that it can be replaced by other options.

Comment (zdroid) [reply pages/common/john.md:28]: Why not just put `md5crypt` here instead?

Comment (agnivade) [reply pages/common/john.md:28]: That's what I am suggesting to do here :) Perhaps the way I put it across might have meant something else.

File: pages/common/john.md [hunk 2]
```diff
@@ -25,12 +25,12 @@
...
```
Comment (zdroid) [inline pages/common/john.md:28]:
```suggestion
`john --format={{md5crypt}} {{path/to/hashes.txt}}`
```

Comment (dhondta) [reply pages/common/john.md:28]: @zdroid I think @agnivade  is right ; `hash_format` is more generic and anyway, another example is provided for listing possible hash formats. So, specifying a particular hash format seems confusing, I think.

File: pages/common/john.md (hunk 1 above)
Comment (sbrl) [reply pages/common/john.md:28]: Ah, I see. I must have misinterpreted. I'm somewhat cautious of including `md5crypt` on it's own here, sincce md5 is known to be a weak hashing algorithm - and this is a password-based tool. But perhaps it will be ok because this is a password cracking tool, rather than a hashing.

File: pages/common/john.md (hunk 2 above)
Comment (sbrl) [reply pages/common/john.md:28]: @dhondta so you're saying that we should leave it as `hash_format`? I'd agree with that, but just want to make sure :P

Comment (dhondta) [reply pages/common/john.md:28]: @sbrl Yes, that's indeed what I'm saying. Let us leave it as `hash_format`.

Comment (agnivade) [reply pages/common/john.md:28]: But that's not what I said :) So your statement "agnivade is right" is not right :laughing: 

The reason I suggested to use `md5crypt` is because we prefer concrete examples rather than generic ones to clarify the command usage, and in fact, it is part of our style guide.

I am wondering why you think specifying a hash format can be confusing ? Yes, there is an example to list hash formats, and a concrete example can stay too.

The discussion whether md5crypt is an appropriate example or not can of course be debated, and if needed something else can be used. But following our usual guidelines, I am in favor of using a concrete example.

Comment (zdroid) [reply pages/common/john.md:28]: @agnivade This looks like a case when two rights make one wrong. :laughing:

I'm for md5crypt being put here, as there is `--format` preceding it.

Comment (sbrl) [reply pages/common/john.md]: Oops, I totally forgot that I'm the commit author here lol

PR updated
